src/expansion/trim_expanded_variable.c

```diff
@@ -35,7 +35,7 @@ char	*trim_expanded_variable(t_string *expand_info, t_ctx *ctx)
 	char	*result;
 
 	consume_char(expand_info);
-	if (*expand_info->right == '\0')
+	if (*expand_info->right == '\0' || ft_strchr(" \t", *expand_info->right))
```

Comment:
The condition `*expanded_info->right == '\0'` fails to handle quoted variables correctly.
For example `echo "$HOME$"`

We could also consider preserving '$' when followed by non-identifier characters (e.g., `echo $/` should display '$/')

``` bash
bash-5.2$ echo $/
$?
bash-5.2$ echo $:
$:
```